starboard/xb1/shared/gpu_base_video_decoder.cc

```diff
@@ -474,7 +474,6 @@ void GpuVideoDecoderBase::Reset() {
     decoder_thread_->job_queue()->Schedule(
         std::bind(&GpuVideoDecoderBase::DrainDecoder, this));
     decoder_thread_.reset();
-    SB_DCHECK(decoder_behavior_.load() == kDecodingStopped);

```

Comment:
If we don't remove this SB_DCHECK(), will it fail?

Reply:
It will. Basically, this function first sets state to kResettingDecoder, than call DrainDecoder(), than this check.
DrainDecoder() is supposed to set state to kDecodingStopped, but it will not if is_drain_decoder_called_ is set.